test/letters/actions/letters.unit.spec.js

```diff
@@ -1,86 +1,495 @@
+    type: ADDRESS_TYPES.military,
+    city: 'apo',
+    state: 'secret'
+  };
 
+  beforeEach(setup);
+  afterEach(teardown);
 
-// Skipping these for now because we're having trouble with making a global way to "mock"
-//  apiRequest(). The answer to this might just be more copy pasta right here to setup
-//  like we've done elsewhere (schemaform save-load-actions, if I recall correctly).
-describe.skip('getMailingAddress', () => {
-  after(() => {
-    resetFetch();
+  it('dispatches SAVE_ADDRESS_PENDING first', (done) => {
+    const thunk = saveAddress(frontEndAddress);
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(SAVE_ADDRESS_PENDING);
+      }).then(done, done);
   });
 
-  it('should transform military address fields to generic ones', (done) => {
-    mockApiRequest(backendAddress);
+  it('dispatches SAVE_ADDRESS_SUCCESS on update success', (done) => {
+    const thunk = saveAddress(frontEndAddress);
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.secondCall.args[0];
+        expect(action.type).to.equal(SAVE_ADDRESS_SUCCESS);
+        expect(action.address).to.eql(frontEndAddress);
+      }).then(done, done);
+  });
 
-    // Had to do the expect()s inside the spy's function because it's asynchronous
-    const dispatchSpy = sinon.spy((action) => {
-      const address = action.data;
-      try {
-        expect(address.city).to.equal(backendAddress.militaryPostOfficeTypeCode);
-        expect(address.state).to.equal(backendAddress.militaryStateCode);
-        expect(address.militaryPostOfficeTypeCode).to.be.undefined;
-        expect(address.militaryStateCode).to.be.undefined;
+  it('dispatches SAVE_ADDRESS_FAILURE on update failure', (done) => {
+    global.fetch.returns(Promise.reject({}));
+    const thunk = saveAddress(frontEndAddress);
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.secondCall.args[0];
+        expect(action.type).to.equal(SAVE_ADDRESS_FAILURE);
+      }).then(done, done);
+  });
+});
 
-        done();
-      } catch (error) {
-        done(error);
-      }
+describe('getLettersList', () => {
+  beforeEach(setup);
+  afterEach(teardown);
+
+  const lettersResponse = {
+    data: {
+      attributes: {
+        letters: [
+          { name: 'Proof of Service Letter', letterType: 'proof_of_service' },
+          { name: 'Civil Service Preference Letter', letterType: 'civil_service' }
+        ],
+        fullName: 'Mark Webb'
+      },
+      id: null,
+      type: 'evss_letters_letters_response'
+    }
+  };
+
+  it('dispatches GET_LETTERS_SUCCESS when GET succeeds', (done) => {
+    global.fetch.returns(Promise.resolve({
+      headers: { get: () => 'application/json' },
+      ok: true,
+      json: () => Promise.resolve(lettersResponse)
```

Comment:
🤔  We can't write a helper function to mock `fetch`, but we can make a helper function to mock what `fetch` _returns_.

Something like:
```js
function craftApiResponse(json, ok = true) {
  const resolveOrReject = ok ? Promise.resolve : Promise.reject;
  return resolveOrReject({
    headers: { get: () => 'application/json' },
    ok,
    json: () => resolveOrReject(json)
  });
}

...

global.fetch.returns(craftApiResponse(lettersResponse));
```

Reply:
This makes total sense, but was running into a strange issue with my implementation so decided to push this off to another PR that will also probably address pdf download tests.